flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/JobMasterPartitionTrackerImpl.java

```diff
@@ -120,6 +125,26 @@ public Collection<ResultPartitionDeploymentDescriptor> getAllTrackedPartitions()
         return partitionInfos.values().stream().map(PartitionInfo::getMetaInfo).collect(toList());
     }
 
+    @Override
+    public void connectToResourceManager(ResourceManagerGateway resourceManagerGateway) {
+        this.resourceManagerGateway = resourceManagerGateway;
+    }
+
+    @Override
+    public List<ShuffleDescriptor> getClusterPartitionShuffleDescriptors(
+            IntermediateDataSetID intermediateDataSetID) {
+        Preconditions.checkNotNull(
+                resourceManagerGateway, "JobMaster is not connected to ResourceManager");
+        try {
+            return this.resourceManagerGateway
```

Comment:
Can we add a cache for it? Otherwise it can result in thousands of RPCs when deploying a large scale job vertex.

Reply:
A cache of the shuffle descriptors of cluster partitions is added. [4bfe582](https://github.com/apache/flink/pull/19653/commits/4bfe582b1f8d8c60bb41663e93699c988ac66ed3)